### DISCUSSION TITLE
Which notebook is the primary one for single stock trading?

### DISCUSSION THREAD

**Discussion by Zymoo** [AI4Finance-Foundation/FinRL#201] [Q&A]
Hi, 

I would like to ask you which one of your notebooks is the most recent one regarding single stock trading?
Currently there are two as such under FinRL-Library/notebooks/old:
 * FinRL_single_stock_trading - which uses predefined StockTradingEnv 
 * DRL_single_stock_trading.ipynb - where you define another custom environment

Additionally, I read your [blogpost](https://towardsdatascience.com/finrl-for-quantitative-finance-tutorial-for-single-stock-trading-37d6d7c30aac) where you mention that "**We can’t use an environment for stock trading to trade bitcoin, and vice versa**.". I do not quite understand why is that? The strategy behind both seems quite similar to me and I planned to train and test bitcoin trading using your predefined environments.

I would greatly appreciate any feedback of yours!

**Comment by b-tank1** [comment]
I've tried both. They both pretty much work, though in the FinRL I am seeing that the trained model actually underperforms the buy-and-hold benchmark using the given stock ('AAPL') and time range (train jan 2009-dec2018, test >=2019). Not sure if anyone else found the same.

**Reply by Zymoo** [reply]
I did run into a few issues with some functions being changed into classes and the other way around. Both notebooks seem a bit outdated therefore was my question.

**Comment by jman563412** [comment]
I'd assume that by "we can't use an environment for stock trading to trade bitcoin" etc. has to do with the lookback period as well as the timesteps. I'd assume if you modify the variables appropriately there's nothing wrong with doing that.

**Reply by jman563412** [reply]
PS: There's 252 stock and forex trading days in a year, while crypto trading days in a year is 365 (every day)

**Reply by Zymoo** [reply]
I agree, this is what I've been thinking before. Happy to hear similar thoughts 👍

**Reply by jman563412** [reply]
Actually, I retract my earlier statement. I've tried using all three notebooks to trade only one stock yet I can't seem to stop running into dependency errors or type conversion errors between numpy and pandas. If you have a working version of the single-stock implementation I'm dying to see it (literally, have a project due in 2 days).
